### PR TITLE
Tests for pathologcial jwk cases

### PR DESCRIPTION
**Pull request by mystenmark**
Tests for handling duplicated / conflicting JWKs.

These cases are unlikely to happen in practice, as oauth providers will almost certainly remain well-behaved. Nevertheless, we want to continue behaving correctly even with pathological inputs.

### CODE REVIEW

**Review by lxfind — APPROVED** [review]

File: crates/sui-benchmark/tests/simtest.rs [hunk 1]
```diff
@@ -405,6 +406,26 @@ mod test {
         default_num_validators: usize,
         default_epoch_duration_ms: u64,
     ) -> TestCluster {
+        sui_node::set_jwk_injector(Arc::new(|_authority, provider| {
```
Comment (lxfind) [inline crates/sui-benchmark/tests/simtest.rs:409]: Why is this needed by the benchmark?

Comment (mystenmark) [reply crates/sui-benchmark/tests/simtest.rs:409]: It's just so that the authenticator state code is running during theses tests, especially crash and restart tests

Comment (lxfind) [reply crates/sui-benchmark/tests/simtest.rs:409]: Should we put it in the test cluster code instead of here so it has more coverage?
